pybossa/cache/projects.py

```diff
@@ -174,20 +175,29 @@ def search_lock_status_sorting_result():
                 task.worker_filter,
                 task.worker_pref
                 FROM task
-                WHERE task.project_id =:project_id""" + filters +\
-                " ORDER BY %s" % (args.get('order_by') or 'priority_0 desc')
+                WHERE task.project_id =:project_id"""
 
         params["assign_user"] = args["sql_params"]["assign_user"]
+
+        sql_without_reserve_filter = sql + filters +\
+                " ORDER BY %s" % (args.get('order_by') or 'priority_0 desc')
+        tasks_without_reserve_filter = [row for row in session.execute(text(sql_without_reserve_filter), params)]
+
+        task_reserve_filter = args.get("filter_by_wfilter_upref", {}).get("reserve_filter", "")
+        if task_reserve_filter:
+            sql_with_reserve_filter = sql + filters + task_reserve_filter
+            tasks_with_reserve_filter = session.execute(text(sql_with_reserve_filter), params)
+        else:
+            tasks_with_reserve_filter = tasks_without_reserve_filter
+        task_id_with_reserve_filter = set([row.id for row in tasks_with_reserve_filter])
+
```

Comment:
I wanted to reduce DB executions by checking if "reserve_filter" is empty, if it is empty, we do not need to make DB calls.

Reply:
In case of PR code, there would be two db executions for reserve filters; one at line 184 and another at 189. In proposed code, there would be only one db execution always.

Reply:
> In proposed code, there would be only one db execution always.

we need to query db twice, one with reserve_filter, one without. I do not understand why there would be one db execution always.

Reply:
discussed w/ weiran the code changes from code maintainability perspective.